### ISSUE TITLE
Copy a file

### ISSUE THREAD

**Issue by Xkonti** [Xkonti/file-api#26]
An endpoint that copies a specified file to a specified path. This can also cause renaming - it's an equivalent of Linux's `cp` command.

- Verb: `POST`
- Path: `/file/copy`

Query parameters:
| Name | Value | Default value | Description |
| --- | --- | --- | --- |
| `source` | string | **required** | A path to a file which should be copy. |
| `destination` | string | **required** | A new path for the file. |
| `overwrite` | `true` or `false` | `false` | Whether to overwrite if there already is a file at the destination path. |

Responses:
- `204 No Content` - when the file was successfully copied.
- `404 Not Found` - when couldn't find the file or if the source is pointing to a directory.
- `409 Conflict` - when there already is a file at the destination path but `overwrite` is set to false. Also if destination is an existing directory.
- `400 Bad Request` - when the path is invalid or when the new path is invalid.

## Extra info
The new path of the file should be also verified with `isPathValid` to make sure the outcome will be what user expects.

User can request to copy file `/docs/private/poetry.md` to `/docs/private/hidden/art.md`. It means that:
1. Directory `hidden` needs to be created.
2. File `poetry.md` needs to be copied as `art.md` to the `hidden` directory.

If the `overwrite` is set to `true` and the destination points to an already existing file, it will be overwritten.

**Comment by sathuhebbar** [comment]
Hello, may I try working on this issue?

Thanks.

**Comment by Xkonti** [comment]
Sure! Go for it 😉

**Comment by sathuhebbar** [comment]
Cool, just had a few questions:

https://github.com/Xkonti/file-api/blob/bd3b6fbfd75e3aea70dae70f8f9d1470f91a2869/src/endpoints/file/getFile.ts#L7

This function seems to have the error message:

https://github.com/Xkonti/file-api/blob/bd3b6fbfd75e3aea70dae70f8f9d1470f91a2869/src/endpoints/file/getFile.ts#L13


in the case that the source file is not found. 
Should the `directory` not be `file` in this error message?

**Comment by Xkonti** [comment]
You're absolutely right! If you want you can include a fix. The best option would be to use a common response like here:

https://github.com/Xkonti/file-api/blob/bd3b6fbfd75e3aea70dae70f8f9d1470f91a2869/src/utils/pathUtils.ts#L2

```ts
return provideValidPathMsg;
```

**Comment by sathuhebbar** [comment]
Cool, I'll work on that separately after adding the copy feature first?

**Comment by sathuhebbar** [comment]
I've tried working on it, please do have a look. I'll improve the error messages and add tests if required.

Thanks!
